server/server.js

```diff
@@ -42,17 +43,44 @@ async function buildSchema (models) {
   }
 
   const dataSources = await models.DataSource.findAll()
-  const dataSourcesJson = dataSources.map((dataSource) => {
+  let dataSourcesJson = dataSources.map((dataSource) => {
     return dataSource.toJSON()
   })
 
   const resolvers = await models.Resolver.findAll({
     include: [models.DataSource]
   })
-  const resolversJson = resolvers.map((resolver) => {
+  let resolversJson = resolvers.map((resolver) => {
     return resolver.toJSON()
   })
 
+  if (_.isEmpty(graphQLSchemaString) || _.isEmpty(dataSourcesJson) || _.isEmpty(resolversJson)) {
```

Comment:
If you look at line 40 we define `let graphQLSchemaString = '{}'`

https://github.com/aerogear/data-sync-server/pull/13/files/7b540109760d1d7e1e57e2b1bb790f8b68f2a7f0..15cb27b90800f444fdfd18e47e9bba590f8bf466#diff-ea75b530d74f94670cea053feb8b961bR40

This means there is a small edge case where the data sources + resolvers are available but no schema and this condition would return truthy. It could lead to unexpected behaviour or crashing

Reply:
Good catch! I've only tried with all tables empty and didn't catch that.